app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go

```diff
@@ -113,7 +113,7 @@ func (l *LoadBalancerSyncer) CreateLoadBalancer(ing *v1beta1.Ingress, forceUpdat
 		return err
 	}
 	// Create health checks to be used by the backend service.
-	healthChecks, hcErr := l.hcs.EnsureHealthCheck(l.lbName, ports, forceUpdate)
+	healthChecks, hcErr := l.hcs.EnsureHealthCheck(l.lbName, ports, client, forceUpdate)
```

Comment:
Note that you are passing go-client for only one cluster. You probably want to send the clients for all clusters.
To make it easier, you can break the changes into 2 stages:
* Make the happy path work: Compute the readiness probe for any one pod in any one cluster and use it in health check. This is easier, you need to pass only one client and fetch any one random pod and use its readiness probe. This works fine for the valid case where readiness probe is same for pods in all clusters.
* Detect the error scenario: Find the readiness probes of all pods for all services in all clusters and generate an error if they dont match. For this, you need to pass all clients, iterate over all of them to fetch pods in all clusters and then compare the readiness probes to generate an error if required. This is to handle error situations gracefully. We should detect the error scenario and generate a clear actionable error message for the user rather than showing them success and then they having to debug why is this not working.

You can directly do both 1 and 2 in this PR or just 1 for now in this PR. Let me know what you are comfortable with.

Reply:
Alright, I think I'll aim for 1 for now and see how it goes. If I have time to work on the next step when that's done I'll just continue here, otherwise I'll wrap up the PR and create an issue for step 2.

Reply:
sounds good. You can add a TODO for second to do it in another PR, if you dont get to it in this PR.